internal/pkg/vault/secrets_test.go

```diff
@@ -865,3 +898,170 @@ func (caller *InMemoryMockCaller) Do(req *http.Request) (*http.Response, error)
 		return nil, errors.New("unsupported HTTP method")
 	}
 }
+
+func TestHttpSecretStoreManager_GetKeys(t *testing.T) {
+	TestConnError := pkg.NewErrSecretStore("testing conn error")
+	TestConnErrorPathNotFound := pkg.NewErrPathNotFound("testing path error")
+	testData := listTestSecretsKeysData()
+	tests := []struct {
+		name              string
+		path              string
+		expectedValues    []string
+		expectError       bool
+		expectedErrorType error
+		expectedDoCallNum int
+		caller            pkg.Caller
+	}{
+		{
+			name:              "Get Key",
+			path:              testPath,
+			expectedValues:    []string{"one", "two", "three/"},
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath],
+			},
+		},
+		{
+			name:              "No keys error",
+			path:              testPath2,
+			expectedValues:    []string{},
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath2],
+			},
+		},
+		{
+			name:              "Subpath",
+			path:              testPath3,
+			expectedValues:    nil,
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath3],
+			},
+		},
+		{
+			name:              "Get non-existent Key",
+			path:              "/one",
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: pkg.NewErrPathNotFound("Does not exist"),
+			expectedDoCallNum: 1,
+			caller: &ErrorMockCaller{
+				ReturnError: false,
+				StatusCode:  404,
+				ErrorType:   pkg.NewErrPathNotFound("Does not exist"),
+			},
+		},
+		{
+			name:              "Get all Keys",
+			path:              "",
+			expectedValues:    []string{"one", "two", "three/"},
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath],
+			},
+		},
+		{
+			name:              "Handle HTTP no path error",
+			path:              testPath,
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: TestConnErrorPathNotFound,
+			expectedDoCallNum: 1,
+			caller: &ErrorMockCaller{
+				ReturnError: false,
+				StatusCode:  404,
+				ErrorType:   pkg.NewErrPathNotFound("Not found"),
+			},
+		},
+		{
+			name:              "Handle non-200 HTTP response",
+			path:              testPath,
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: TestConnError,
+			expectedDoCallNum: 1,
+			caller: &ErrorMockCaller{
+				ReturnError: false,
```

Comment:
Why false when error is specified? Seems to always be false so can be removed?
Applies to other test cases also